### PR TITLE
Lint Cython files

### LINKED ISSUE #3829: Linting for Cython files
see https://github.com/ray-project/ray/pull/3541/#discussion_r250057391

### PR DESCRIPTION
**Pull request by pcmoritz**
fixes https://github.com/ray-project/ray/issues/3829

### CODE REVIEW

File: python/ray/includes/ray_config.pxi [hunk 1]
```diff
@@ -119,7 +122,8 @@ cdef class Config:
 
     @staticmethod
     def node_manager_forward_task_retry_timeout_milliseconds():
-        return RayConfig.instance().node_manager_forward_task_retry_timeout_milliseconds()
+        return RayConfig.instance()\
```
Comment (robertnishihara) [inline python/ray/includes/ray_config.pxi]: don't use `\`, instead do something like

```
return (RayConfig.instance()
        .node_manager_forward_task_retry_timeout_milliseconds())
```

this applies in other places as well.

**Review by guoyuhong — COMMENTED** [review]
In format.sh, there is flake8 call https://github.com/ray-project/ray/blob/master/.travis/format.sh#L54-L57 . Will the adding of `.flake8` affect the existing flake8 call?

File: .travis/format.sh [hunk 2]
```diff
@@ -84,6 +84,9 @@ else
     format_changed
 fi
 
+# Lint Cython files (see .flake8 in the root directory)
+flake8
```
Comment (ericl) [inline .travis/format.sh]: How long does this take to run?

File: python/ray/includes/ray_config.pxi (hunk 1 above)
Comment (pcmoritz) [reply python/ray/includes/ray_config.pxi]: I fixed it for the ones in this file, for some cython constructs we do need to use \ (this is quite standard, see https://github.com/apache/arrow/blob/master/python/pyarrow/_cuda.pyx)

File: .travis/format.sh (hunk 2 above)
Comment (pcmoritz) [reply .travis/format.sh]: I changed this now so only files that were modified are run through flake8, so the runtime is now similar to what it was before (~20 sec).

**Comment by pcmoritz** [comment]
@guoyuhong This is fixed now

File: python/ray/includes/task.pxd [hunk 3]
```diff
@@ -18,10 +18,13 @@ from ray.includes.unique_ids cimport (
 )
 
 
-cdef extern from "ray/raylet/task_execution_spec.h" namespace "ray::raylet" nogil:
-    cdef cppclass CTaskExecutionSpecification "ray::raylet::TaskExecutionSpecification":
+cdef extern from "ray/raylet/task_execution_spec.h" \
+        namespace "ray::raylet" nogil:
+    cdef cppclass CTaskExecutionSpecification \
```
Comment (robertnishihara) [inline python/ray/includes/task.pxd]: Can these `\`'s be removed?

File: python/ray/includes/task.pxd [hunk 4]
```diff
@@ -34,31 +37,35 @@ cdef extern from "ray/raylet/task_spec.h" namespace "ray::raylet" nogil:
     cdef cppclass CTaskArgument "ray::raylet::TaskArgument":
         pass
 
-    cdef cppclass CTaskArgumentByReference "ray::raylet::TaskArgumentByReference":
-        CTaskArgumentByReference(const c_vector[CObjectID] &references);
+    cdef cppclass CTaskArgumentByReference\
```
Comment (robertnishihara) [inline python/ray/includes/task.pxd]: `\`

File: python/ray/includes/common.pxd [hunk 5]
```diff
@@ -92,18 +100,21 @@ cdef extern from "ray/gcs/format/gcs_generated.h" nogil:
...
+# This is a workaround for C++ enum class since Cython has no corresponding
+# representation.
 cdef extern from "ray/gcs/format/gcs_generated.h" namespace "Language" nogil:
     cdef CLanguage LANGUAGE_PYTHON "Language::PYTHON"
     cdef CLanguage LANGUAGE_CPP "Language::CPP"
     cdef CLanguage LANGUAGE_JAVA "Language::JAVA"
 
 
-cdef extern from "ray/raylet/scheduling_resources.h" namespace "ray::raylet" nogil:
+cdef extern from "ray/raylet/scheduling_resources.h" \
```
Comment (robertnishihara) [inline python/ray/includes/common.pxd]: `\`

File: python/ray/includes/libraylet.pxd [hunk 6]
```diff
@@ -37,7 +37,8 @@ cdef extern from "ray/gcs/format/gcs_generated.h" nogil:
         GCSProfileTableDataT()
 
 
-ctypedef unordered_map[c_string, c_vector[pair[int64_t, double]]] ResourceMappingType
+ctypedef unordered_map[c_string, c_vector[pair[int64_t, double]]] \
```
Comment (robertnishihara) [inline python/ray/includes/libraylet.pxd]: `\`

File: python/ray/includes/task.pxd (hunk 3 above)
Comment (pcmoritz) [reply python/ray/includes/task.pxd]: I don't think so, also we shouldn't obsess about removing \.

File: python/ray/includes/task.pxd (hunk 4 above)
Comment (pcmoritz) [reply python/ray/includes/task.pxd]: can't be removed

File: python/ray/includes/common.pxd (hunk 5 above)
Comment (pcmoritz) [reply python/ray/includes/common.pxd]: can't be removed

File: python/ray/includes/libraylet.pxd (hunk 6 above)
Comment (pcmoritz) [reply python/ray/includes/libraylet.pxd]: can't be removed

File: python/ray/includes/task.pxd (hunk 4 above)
Comment (robertnishihara) [reply python/ray/includes/task.pxd]: ok at least add a space before the `\`

Comment (pcmoritz) [reply python/ray/includes/task.pxd]: already done, this diff is outdated

**Comment by raulchen** [comment]
What tool did you use to format cython code?

**Comment by pcmoritz** [comment]
As far as I know there is no tool to do it automatically (in particular, yapf doesn't, see https://github.com/google/yapf/issues/175). I did it by hand.

**Review by raulchen — APPROVED** [review]
LGTM

**Pull request merged**